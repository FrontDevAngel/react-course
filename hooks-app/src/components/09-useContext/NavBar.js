import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
    return (
        <nav className='navbar navbar-expand-sm navbar-dark bg-dark px-3'>
            <Link className='navbar-brand' to='./'>
                UseContext
            </Link>
            <div className='navbar-nav'>
                <NavLink
                    activeClassName='active'
                    className='nav-item nav-link'
                    exact
                    to='./'>
                    Home
                </NavLink>
                <NavLink
                    activeClassName='active'
                    className='nav-item nav-link'
                    exact
                    to='./about'>
                    About
                </NavLink>
                <NavLink
                    activeClassName='active'
                    className='nav-item nav-link'
                    exact
                    to='./login'>
                    Login
                </NavLink>
            </div>
        </nav>
    );
};
