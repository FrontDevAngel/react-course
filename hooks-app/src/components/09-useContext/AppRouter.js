import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
} from 'react-router-dom';
import { NavBar } from './NavBar';

import { HomeScreen } from './HomeScreen';
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>
                <NavBar />
                <div className='container'>
                    <Switch>
                        <Route exact path='/' component={HomeScreen} />
                        <Route exact path='/about' component={AboutScreen} />
                        <Route exact path='/login' component={LoginScreen} />

                        {/* Redirecci√≥n si no cumple con ninguna de las anteriores */}

                        {/* Dirigir al componente ////////////////////// */}
                        {/* <Route component={ FailScreen } /> */}

                        {/* Dirigir al componente existente ////////////////////// */}
                        {/* <Route component={ HomeScreen } /> */}

                        {/* Redirecta a un Path ////////////////////// */}
                        <Redirect to='/' />
                    </Switch>
                </div>
            </div>
        </Router>
    );
};
