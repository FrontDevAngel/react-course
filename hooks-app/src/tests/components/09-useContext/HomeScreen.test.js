import React from 'react';
import { shallow, mount } from 'enzyme';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en <HomeScreen />', () => {
    const user = {
        id: 123456,
        name: '√Ångel Pantoja',
    };

    const setUser = jest.fn();

    const wrapper = mount(
        <UserContext.Provider value={{ user, setUser }}>
            <HomeScreen />
        </UserContext.Provider>,
    );
    it('debe de mostrarste correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
