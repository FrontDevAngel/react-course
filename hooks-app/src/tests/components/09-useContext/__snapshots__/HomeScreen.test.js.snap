// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HomeScreen /> debe de mostrarste correctamente 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "id": 123456,
   "name": "√Ångel Pantoja"
}
  </pre>
</div>
`;
